crates/cargo-codspeed/src/build.rs

```diff
@@ -1,168 +1,171 @@
-            benches_to_build.len(),
-            if all_benches_count > actual_benches_count {
-                format!(
-                    " ({} filtered out)",
-                    all_benches_count - actual_benches_count
-                )
-            } else {
-                "".to_string()
-            }
-        ),
-        &style::TITLE,
-    )?;
-
-    let context = ws.gctx();
-    let codspeed_root_target_dir = get_codspeed_target_dir(ws);
-    // Create and clear packages target directories
-    for package in packages_to_build.iter() {
-        let package_target_dir = codspeed_root_target_dir.join(package.name());
-        create_dir_all(&package_target_dir)?;
-        clear_dir(&package_target_dir)?;
-    }
-    let mut built_benches = 0;
-    for bench in benches_to_build.iter() {
-        ws.gctx().shell().status_with_color(
-            "Building",
-            format!("{} {}", bench.package.name(), bench.target.name()),
-            &style::ACTIVE,
-        )?;
-        let is_root_package = ws.current_opt() == Some(bench.package);
-        let benches_names = vec![bench.target.name()];
-        let compile_opts = get_compile_options(
-            context,
-            &features,
-            &profile,
-            bench.package,
-            benches_names,
-            is_root_package,
-        )?;
-        let result = cargo::ops::compile(ws, &compile_opts)?;
-        let built_units = result
-            .tests
-            .into_iter()
-            .filter(|u| u.unit.target.is_bench())
-            .collect_vec();
-        if built_units.is_empty() {
-            continue;
-        }
-        built_benches += 1;
-        let codspeed_target_dir = codspeed_root_target_dir.join(bench.package.name());
-        for built_unit in built_units.iter() {
-            let bench_dest = codspeed_target_dir
-                .clone()
-                .join(built_unit.unit.target.name());
-            std::fs::copy(built_unit.path.clone(), bench_dest)?;
-        }
-    }
-
-    if built_benches == 0 {
+    if built_benches.is_empty() {
         bail!(
-            "No benchmark target found. \
+            "No benchmark target found.\
             Please add a benchmark target to your Cargo.toml"
         );
     }
-    ws.gctx().shell().status_with_color(
-        "Finished",
-        format!("built {} benchmark suite(s)", built_benches),
-        &style::SUCCESS,
-    )?;
+
+    let codspeed_target_dir = get_codspeed_target_dir(metadata);
+    let built_bench_count = built_benches.len();
+    for built_bench in built_benches {
+        let codspeed_target_package_dir = codspeed_target_dir.clone().join(&built_bench.package);
+
+        std::fs::create_dir_all(&codspeed_target_package_dir)?;
+        std::fs::copy(
+            built_bench.executable_path,
+            codspeed_target_package_dir.join(built_bench.bench),
+        )?;
+    }
+
+    eprintln!("Built {built_bench_count} benchmark suite(s)");
```

Comment:
As discussed, to keep the same behavior as before, we should clean each package target directory before copying into it.